### ISSUE TITLE
Loaded TorchScript Model in C++, and inference taken longer time?

### ISSUE THREAD

**Issue by ciderpark** [pytorch/pytorch#47737]
I converted .pth to .pt, and loaded it in libtorch. Then, I compared the inference time between python and C++, got this:
resnet18 : 29ms(python)->31ms(C++)
resnet50: 66ms->73ms
resnet152: 140ms->223ms
maskrcnn: 326ms->501ms
On GPU: GTX1060
And Torch Script via Annotation method takes longer time than Torch Script via Tracing method in libtorch.
How could this happen?
I also trained yolo-v3 with my libtorch implementation and pytorch implementation, got:
45ms(C++) : 90ms(python), C++ is faster.

cc @gmagogsfm

**Comment by suo** [comment]
Thanks for the report! Can you provide some more information about the environment? e.g. run the environment script suggested by the issue template. 

In addition, it would be helpful to provide code to reproduce the issue, so that we can take a closer look.

**Comment by ciderpark** [comment]
> Thanks for the report! Can you provide some more information about the environment? e.g. run the environment script suggested by the issue template.
> 
> In addition, it would be helpful to provide code to reproduce the issue, so that we can take a closer look.

Hi, I installed pytorch 1.7.0, libtorch 1.7.0 cxx11, CUDA 10.2, and built torchvison c++ API 0.7.0 from source in release with CUDA = ON.
The code is following the tutorial: https://pytorch.org/tutorials/advanced/cpp_export.html.
And Torch Script via Tracing failed on vision.maskrcnn.

**Comment by ciderpark** [comment]
> Thanks for the report! Can you provide some more information about the environment? e.g. run the environment script suggested by the issue template.
> 
> In addition, it would be helpful to provide code to reproduce the issue, so that we can take a closer look.
Hi, there is a interesting performance. I captured a 1920 * 1080 image by camera, inference(mask-rcnn) taken 1193ms the first time, then 636ms the second time, then 9945ms the third time! And for the next several times, it's always about 250ms !!!
How could this happen !?

**Comment by ciderpark** [comment]
> Thanks for the report! Can you provide some more information about the environment? e.g. run the environment script suggested by the issue template.
> 
> In addition, it would be helpful to provide code to reproduce the issue, so that we can take a closer look.

Hi, there is a interesting performance. I captured a 1920 * 1080 image by camera, inference(mask-rcnn) taken 1193ms the first time, then 636ms the second time, then 9945ms the third time! And for the next several times, it's always about 250ms !!!
How could this happen !?

**Comment by ciderpark** [comment]
> Thanks for the report! Can you provide some more information about the environment? e.g. run the environment script suggested by the issue template.
> 
> In addition, it would be helpful to provide code to reproduce the issue, so that we can take a closer look.

Hi, there is a interesting performance. I captured a 1920 * 1080 image by camera, inference(mask-rcnn) taken 636ms the first time,  then 9945ms the second time! And for the next several times, it's always about 250ms !!!
How could this happen !?

**Comment by SeanChanCXG** [comment]
> I converted .pth to .pt, and loaded it in libtorch. Then, I compared the inference time between python and C++, got this:
> resnet18 : 29ms(python)->31ms(C++)
> resnet50: 66ms->73ms
> resnet152: 140ms->223ms
> maskrcnn: 326ms->501ms
> On GPU: GTX1060
> And Torch Script via Annotation method takes longer time than Torch Script via Tracing method in libtorch.
> How could this happen?
> I also trained yolo-v3 with my libtorch implementation and pytorch implementation, got:
> 45ms(C++) : 90ms(python), C++ is faster.
> 
> cc @gmagogsfm

Have you solved this problem？ I have the same problem that cxx inference is slower than python

**Comment by ciderpark** [comment]
> > I converted .pth to .pt, and loaded it in libtorch. Then, I compared the inference time between python and C++, got this:
> > resnet18 : 29ms(python)->31ms(C++)
> > resnet50: 66ms->73ms
> > resnet152: 140ms->223ms
> > maskrcnn: 326ms->501ms
> > On GPU: GTX1060
> > And Torch Script via Annotation method takes longer time than Torch Script via Tracing method in libtorch.
> > How could this happen?
> > I also trained yolo-v3 with my libtorch implementation and pytorch implementation, got:
> > 45ms(C++) : 90ms(python), C++ is faster.
> > cc @gmagogsfm
> 
> Have you solved this problem？ I have the same problem that cxx inference is slower than python

Hi, this strange performance is caused by JIT(just in time). It worked well after I run the inference in a loop.
